monai/losses/nacl_loss.py

```diff
@@ -0,0 +1,279 @@
+    x_coord = torch.arange(ksize)
+    x_grid_2d = x_coord.repeat(ksize).view(ksize, ksize)
+    x_grid = x_coord.repeat(ksize * ksize).view(ksize, ksize, ksize)
+    y_grid_2d = x_grid_2d.t()
+    y_grid = y_grid_2d.repeat(ksize, 1).view(ksize, ksize, ksize)
+    z_grid = y_grid_2d.repeat(1, ksize).view(ksize, ksize, ksize)
+    xyz_grid = torch.stack([x_grid, y_grid, z_grid], dim=-1).float()
+    mean = (ksize - 1) / 2.0
+    variance = sigma**2.0
+    gaussian_kernel: torch.Tensor = (1.0 / (2.0 * math.pi * variance + 1e-16)) * torch.exp(
+        -torch.sum((xyz_grid - mean) ** 2.0, dim=-1) / (2 * variance + 1e-16)
+    )
+    gaussian_kernel = gaussian_kernel / torch.sum(gaussian_kernel)
+    return gaussian_kernel
+
+
+class GaussianFilter(torch.nn.Module):
+    def __init__(self, dim: int = 3, ksize: int = 3, sigma: float = 1.0, channels: int = 0) -> None:
+        super(GaussianFilter, self).__init__()
+
+        self.svls_kernel: torch.Tensor
+        self.svls_layer: Any
+
+        if dim == 2:
+            gkernel = get_gaussian_kernel_2d(ksize=ksize, sigma=sigma)
+            neighbors_sum = (1 - gkernel[1, 1]) + 1e-16
+            gkernel[int(ksize / 2), int(ksize / 2)] = neighbors_sum
+            self.svls_kernel = gkernel / neighbors_sum
+
+            svls_kernel_2d = self.svls_kernel.view(1, 1, ksize, ksize)
+            svls_kernel_2d = svls_kernel_2d.repeat(channels, 1, 1, 1)
+            padding = int(ksize / 2)
+
+            self.svls_layer = torch.nn.Conv2d(
+                in_channels=channels,
+                out_channels=channels,
+                kernel_size=ksize,
+                groups=channels,
+                bias=False,
+                padding=padding,
+                padding_mode="replicate",
+            )
+            self.svls_layer.weight.data = svls_kernel_2d
+            self.svls_layer.weight.requires_grad = False
+
+        if dim == 3:
+            gkernel = get_gaussian_kernel_3d(ksize=ksize, sigma=sigma)
+            neighbors_sum = 1 - gkernel[1, 1, 1]
+            gkernel[1, 1, 1] = neighbors_sum
+            self.svls_kernel = gkernel / neighbors_sum
+
+            svls_kernel_3d = self.svls_kernel.view(1, 1, ksize, ksize, ksize)
+            svls_kernel_3d = svls_kernel_3d.repeat(channels, 1, 1, 1, 1)
+            padding = int(ksize / 2)
+
+            self.svls_layer = torch.nn.Conv3d(
+                in_channels=channels,
+                out_channels=channels,
+                kernel_size=ksize,
+                groups=channels,
+                bias=False,
+                padding=padding,
+                padding_mode="replicate",
+            )
+            self.svls_layer.weight.data = svls_kernel_3d
+            self.svls_layer.weight.requires_grad = False
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        svls_normalized: torch.Tensor = self.svls_layer(x) / self.svls_kernel.sum()
+        return svls_normalized
+
+
+class MeanFilter(torch.nn.Module):
```

Comment:
And for this class, would it be better to move it under monai/networks/layers/simplelayers.py?